sui_core/src/authority/temporary_store.rs

```diff
@@ -345,10 +346,14 @@ impl<S: BackingPackageStore> ModuleResolver for AuthorityTemporaryStore<S> {
     type Error = SuiError;
     fn get_module(&self, module_id: &ModuleId) -> Result<Option<Vec<u8>>, Self::Error> {
         let package_id = &ObjectID::from(*module_id.address());
+        let package_obj;
         let package = match self.read_object(package_id) {
             Some(object) => object,
             None => match self.package_store.get_package(package_id)? {
-                Some(object) => object,
+                Some(object) => {
+                    package_obj = object;
+                    &package_obj
```

Comment:
Is this the same as `&object.clone()`?

Reply:
No, the `object` here is already owned (`object: Object`), so the clone wouldn't do anything. This is a trick for getting around the lifetimes. If you try doing `Some(object) => &object`, it complains that the reference will not live long enough. This is a trick that forces the lifetime outside of the match, thus letting it live long enough.